File: scalang/src/main/java/com/kapilkoju/autopos/kernel/domain/AbstractBaseEntity.java
package com.kapilkoju.autopos.kernel.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.MappedSuperclass;

@MappedSuperclass
public abstract class AbstractBaseEntity
    extends AbstractEntity<Long>
    implements BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @JsonIgnore
    @Override
    public final Long getIdentity() {
        return getId();
    }

    @Override
    public Long getId() {
        return id;
    }

    public void setId(final Long id) {
        this.id = id;
    }

}

File: kotlang/src/main/kotlin/com/kapilkoju/autopos/kernel/domain/AbstractBaseEntity.kt
package com.kapilkoju.autopos.kernel.domain

import com.fasterxml.jackson.annotation.JsonIgnore

import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.MappedSuperclass

@MappedSuperclass
abstract class AbstractBaseEntity : AbstractEntity<Long>(), BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private var id: Long? = null

    @JsonIgnore
    override fun getIdentity(): Long? {
        return getId()
    }

    override fun getId(): Long? {
        return id
    }

    fun setId(id: Long?) {
        this.id = id
    }

}
